src/field/pokemon-move.ts

```diff
@@ -30,14 +32,35 @@ export class PokemonMove {
    */
   public maxPpOverride?: number;
 
-  constructor(moveId: MoveId, ppUsed: number = 0, ppUp: number = 0, virtual: boolean = false, maxPpOverride?: number) {
+  constructor(
+    pokemon: Pokemon | null,
+    moveId: MoveId,
+    ppUsed: number = 0,
+    ppUp: number = 0,
+    virtual: boolean = false,
+    maxPpOverride?: number,
```

Comment:
I don't think we should be adding new pseudo-optional params. Since you have to update every call site already, it's better to just do it "properly" instead.
```suggestion
  constructor(
    moveId: MoveId,
    { pokemon, ppUsed = 0, ppUp = 0, virtual = false, maxPpOverride }: {
      pokemon?: Pokemon;
      ppUsed?: number;
      ppUp?: number;
      virtual?: boolean;
      maxPpOverride?: number;
    } = {}
```